Lib/typing.py

```diff
@@ -2294,6 +2296,15 @@ def __subclasscheck__(cls, other):
 
     __instancecheck__ = __subclasscheck__
 
+    def __getitem__(cls, params):
+        if not isinstance(params, tuple):
+            params = (params,)
+        msg = "Parameters to generic types must be types."
+        params = tuple(_type_check(p, msg) for p in params)
+        nparams = len(cls.__dict__.get("__parameters__", ()))
+        _check_generic(cls, params, nparams)
+        return types.GenericAlias(cls, params)
```

Comment:
I think there are no need to define `__getitem__` method as far as `Generic` class has `__class_getitem__` method.

Reply:
I had two issues by not overriding the default:

1. Dunder attributes are not proxied
```
class A(TypedDict, Generic[T]):
    foo: T

assert(A[str].__required_keys__  == frozenset(['foo']) # Raises Attribute error
assert(A[str].__origin__.__required_keys__  == frozenset(['foo']) # Works
```

2. May be a bug in the original code. But even without the changes in this PR, following is valid at runtime (digging in what I think happens is that it inherits a `__get_item__` from `dict` meta):
```
class G(TypedDict):
    g: str

print(G[int]) # G is subscriptable although not a Generic
```

Reply:
@Fidget-Spinner What is your opinion about issue mentioned by Samodya?

Reply:
For 1st issue:
That happens because `_BaseGenericAlias.__getattr__` intentionally doesn't proxy dunders to `self.__origin__` https://github.com/python/cpython/blob/main/Lib/typing.py#L966. I don't know what's the correct behavior until I read the final version of the proposed PEP update/ PEP. Sorry.


For the 2nd behavior:
Since python 3.9, builtin generics support subscripting thanks to PEP 585. E.g. ``dict[str, int]``. Since subclassing TypedDict subclasses `dict`, it also pulls `dict.__class_getitem__` (and hence the type supports arbitrary subscripting). A quick way to verify this:
```python
>>> type(G[int])
<class 'types.GenericAlias'>
>>> G.__mro__
(<class '__main__.G'>, <class 'dict'>, <class 'object'>)
```

Reply:
Re 1: Dunder proxying behavior isn't specified in the PEP draft either :) I switched gears to add this feature through a bpo and implementation as I am not sure whether this change needed a PEP/PEP update (my reasoning in the ticket description). By any chance do you know a PEP that specifies the behavior of `GenericAlias` for another type?

Re 2: I see. Thank you. But that does seem like something that shouldn't be allowed for a non generic Typeddict (Not a big deal though since it doesn't really do anything).

Reply:
Considering following is the behavior for normal Generic class, I will revert the overriding of `__getitem__` to try to proxy the dunder attributes.
```
class A(Generic[T]):
    a: T

print(A.__annotations__) # {'a': ~T}
print(A[str].__annotations__) # raises AttributeError
print(A[str].__origin__.__annotations__) # {'a': ~T}
```